src/sentry/static/sentry/app/components/avatar.jsx

```diff
@@ -23,24 +22,41 @@ class Avatar extends React.Component {
   constructor(...args) {
     super(...args);
     this.state = {
+      gravatarUrl: null,
       showBackupAvatar: false,
       loadError: false,
     };
   }
 
+  componentDidMount() {
+    this.buildGravatarUrl();
+  }
+
+  componentWillReceiveProps() {
+    this.buildGravatarUrl();
+  }
+
   buildGravatarUrl = () => {
-    let url = ConfigStore.getConfig().gravatarBaseUrl + '/avatar/';
+    let avatarType = this.getAvatarType();
 
-    url += MD5(this.props.user.email.toLowerCase());
+    if (avatarType !== 'gravatar') return;
 
-    let query = {
-      s: this.props.size || undefined,
-      d: this.props.default || 'blank',
-    };
+    import(/*webpackChunkName: "crypto-js/md5"*/ 'crypto-js/md5').then(MD5 => {
```

Comment:
`import` won't fire HTTP requests for this resource twice, right?

Reply:
Correct